tests/jballerina-unit-test/src/test/resources/test-src/annotations/external_dependency_annotation.bal

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+// Functions
+@java:ExternalDependency
+function foo(int x) returns int {
+    return x * 2;
+}
+
+function bar() returns string {
+    return "bar";
+}
+
+// Attached functions
+class CorgeClass {
+
+    @java:ExternalDependency
+    function externalCorge() returns string {
+        return "external corge";
+    }
+
+    function corge() returns string {
+        return "corge";
+    }
+}
+
+// Classes
+@java:ExternalDependency
+class BazClass {
+    int x = 0;
+    string y = "";
+}
+
+class QuzClass {
+    int x = 0;
+    string y = "";
+}
+
+// Records
+@java:ExternalDependency
+type FooRec record {
+    int x;
+    string y;
+};
+
+type BarRec record {
+    int x;
+    string y;
+};
+
+// Type Definitions
+@java:ExternalDependency
+type FooType int|string;
+
+type BarType int|string;
+
+// Edge cases
+@deprecated
+@java:ExternalDependency
+@tainted
+type CorgeType int|string;
```

Comment:
AFAIK, you cannot create types using the ValueCreator for these scenarios (only allowed for records and objects). Although the annotation is supposed to work, it has no value for your use case. IMO, you do not need to bear the additional overhead of restricting this annotation only for records and object types, as the annotation attachment has a broader scope than this. However, you should be aware that you only need to preserve record and object types with this annotation, as this annotation on other types has no meaning.

Reply:
Yes you are correct. I added the test cases to verify the ExternalDependency annotation is working as we defined it. Not being able to limit it to records and objects is a limitation on our end. I will discuss this with @hasithaa and update here.

Reply:
This is a generic feature, so technically, it should be able to use it on every function, class, and type, irrespective of runtime limitations.